projects/igniteui-angular/src/lib/select/select.component.ts

```diff
@@ -119,6 +122,28 @@ export class IgxSelectComponent extends IgxDropDownComponent implements IgxSelec
      *
      */
     @Input() public placeholder;
+
+    private _required = false;
+    /**
+     * Sets/gets whether item selection is required.
+     * If not set, `required` will have value `false`.
+     * ```html
+     * <igx-select [required] = "true"></igx-select>
+     * ```
+     * ```typescript
+     * let isRequired =  this.select.required;
+     */
+    public get required() {
+        return this._required;
+    }
+    @Input()
+    public set required(value: boolean) {
```

Comment:
Actually.. Do we even need to expose `[required]` on the select?
It is technically already handled by the [`RequiredValidator`](https://angular.io/api/forms/RequiredValidator) directive from @angular/forms.

Might be easier to not have the Input, than explaining the validation requires an ngControl anyway. Or it might be less discoverable that way?

// CC @rkaraivanov Thoughts?

Reply:
@damyanpetev 

> Actually.. Do we even need to expose [required] on the select?
We don't

Reply:
This wouldn't work with reactive forms anyway. And here's a [sample showing the `ngModel` validator directive ](https://stackblitz.com/edit/angular-newbha?file=src%2Fapp%2Fselect-sample-3%2Fselect-sample-3.component.html) work just fine without this.
@IvayloG You can just drop the property.
Here's the related bit from the input checking the validators:
https://github.com/IgniteUI/igniteui-angular/blob/790dd199c9ccefdb323b8275ca42616d18844425/projects/igniteui-angular/src/lib/directives/input/input.directive.ts#L197-L201

The only thing that bugs me is detecting the change of the validator itself (i.e. runtime required change for the (*) indicator). Probably some validatorsChange subscription somewhere.